contracts/strategies/cpmm/base/CPMMBaseRebalanceStrategy.sol

```diff
@@ -44,13 +45,15 @@ abstract contract CPMMBaseRebalanceStrategy is BaseRebalanceStrategy, CPMMBaseLo
 
     /// @dev See {BaseRebalanceStrategy-_calcCollateralPostTrade}.
     function _calcCollateralPostTrade(int256[] memory deltas, uint128[] memory tokensHeld, uint128[] memory reserves) internal override virtual view returns(uint256 collateral) {
+        uint256 minCollateral = calcInvariant(address(0), tokensHeld) * (tradingFee1 + (tradingFee2 - tradingFee1) / 2)/ tradingFee2;
         if(deltas[0] > 0) {
             collateral = _calcCollateralPostTradeStaticCall(uint256(deltas[0]), tokensHeld[0], tokensHeld[1], reserves[0], reserves[1]);
         } else if(deltas[1] > 0) {
             collateral = _calcCollateralPostTradeStaticCall(uint256(deltas[1]), tokensHeld[1], tokensHeld[0], reserves[1], reserves[0]);
         } else {
             collateral = calcInvariant(address(0), tokensHeld);
         }
+        if(collateral < minCollateral) revert LowPostTradeCollateral();
```

Comment:
Yeah I think it should still be valid. If there are not tradingFees and deltas[0] and deltas[1] are both <= 0 (no rebalancing), then collateral = minCollateral which is fine. So we don't revert. We only revert if collateral < minCollateral because of trading fees when rebalancing very close to current market price. If there are no trading fees and you have to rebalance then postTradeCollateral > preTradeCollateral. But if there are trading fees and you have to rebalance and the price was very close to the current market price postTradeCollateral < preTradeCollateral is possible. I'm giving it some room which is equivalent to the trading fee on one side of the rebalancing quantity.

Using fuzzing this has only failed when one of the reserve quantities in the CFMM is 3 digits or less.